argocd-mixin/dashboards/argocd-overview.json

```diff
@@ -0,0 +1,2932 @@
+          "format": "time_series",
+          "intervalFactor": 1,
+          "legendFormat": "__auto",
+          "range": true,
+          "refId": "A"
+        }
+      ],
+      "title": "Goroutines",
+      "transparent": true,
+      "type": "timeseries"
+    },
+    {
+      "datasource": {
+        "type": "prometheus",
+        "uid": "grafanacloud-prom"
+      },
+      "description": "GC Time Quantiles by ArgoCD Server",
+      "fieldConfig": {
+        "defaults": {
+          "color": {
+            "mode": "palette-classic"
+          },
+          "custom": {
+            "axisCenteredZero": false,
+            "axisColorMode": "text",
+            "axisLabel": "",
+            "axisPlacement": "auto",
+            "barAlignment": 0,
+            "drawStyle": "line",
+            "fillOpacity": 10,
+            "gradientMode": "none",
+            "hideFrom": {
+              "legend": false,
+              "tooltip": false,
+              "viz": false
+            },
+            "lineInterpolation": "linear",
+            "lineWidth": 1,
+            "pointSize": 5,
+            "scaleDistribution": {
+              "type": "linear"
+            },
+            "showPoints": "never",
+            "spanNulls": true,
+            "stacking": {
+              "group": "A",
+              "mode": "none"
+            },
+            "thresholdsStyle": {
+              "mode": "off"
+            }
+          },
+          "mappings": [],
+          "thresholds": {
+            "mode": "absolute",
+            "steps": [
+              {
+                "color": "green",
+                "value": null
+              },
+              {
+                "color": "red",
+                "value": 80
+              }
+            ]
+          },
+          "unit": "short"
+        },
+        "overrides": []
+      },
+      "gridPos": {
+        "h": 8,
+        "w": 12,
+        "x": 12,
+        "y": 95
+      },
+      "id": 38,
+      "links": [],
+      "options": {
+        "legend": {
+          "calcs": [],
+          "displayMode": "list",
+          "placement": "bottom",
+          "showLegend": false
+        },
+        "tooltip": {
+          "mode": "multi",
+          "sort": "none"
+        }
+      },
+      "pluginVersion": "10.0.0-cloud.3.b04cc88b",
+      "targets": [
+        {
+          "datasource": {
+            "uid": "$datasource"
+          },
+          "editorMode": "code",
+          "expr": "go_gc_duration_seconds{job=\"integrations/argocd/argocd-server\", quantile=\"1\"}",
+          "format": "time_series",
+          "intervalFactor": 2,
+          "legendFormat": "__auto",
+          "range": true,
+          "refId": "A"
+        }
+      ],
+      "title": "GC Time Quantiles",
+      "transparent": true,
+      "type": "timeseries"
+    }
+  ],
+  "refresh": "",
+  "schemaVersion": 38,
+  "style": "dark",
+  "tags": [
+    "argocd-integration"
+  ],
+  "templating": {
```

Comment:
Best practice is to have a template variable for the `job` and `instance` label, allowing you to filter the dashboard to particular instances of the SUO.

We may need to discuss this as I see that you have a couple different job labels in use, and may be using them to target specific processes.

Reply:
ArgoCD has quite a few services, I opted to use a different job label for each in my agent scrape config. I can make it scrape all under the same job label although it might not be the best way in my opinion. Open to suggestions here

Reply:
Ahh, on closer inspection, I see that you are filtering by namespace, and you have a "grouping" option as well.

So, in a case where there was more than one "instance" of ArgoCD in a k8s cluster, they would likely be deployed into different namespaces, and therefore you would be able to filter that way?

The grouping seems to further narrow down which ArgoCD instance you're observing, yes?

If so, the approach you're using is fine, and job/instance are not necessary. When we build the Grafana Cloud integration (or k8s app instructions) we'll want to be explicit about how to setup the scrape jobs to match what the mixin is expecting.

Reply:
If I'm correct about my assertions above, the following `.lint` file will allow the linter to pass, even with rules pertaining to job and instance failing. Just put it in a file named `.lint` at the root of the project.

```exclusions:
  template-instance-rule:
    reason: ArgoCD instance filtering is accomplished by the namespace and grouping template variables
  template-job-rule:
    reason: ArgoCD instance filtering is accomplished by the namespace and grouping template variables    
  target-instance-rule:
    reason: ArgoCD instance filtering is accomplished by the namespace and grouping template variables
  target-job-rule:
    reason: ArgoCD instance filtering is accomplished by the namespace and grouping template variables
  panel-units-rule:
    reason: Panel units are self descriptive
```